api/scpca_portal/test/management/commands/test_load_data.py

```diff
@@ -966,7 +1045,7 @@ def test_spatial_metadata(self):
         # │       ├── tissue_lowres_image.png
         # │       └── tissue_positions_list.csv
         # └── spatial_metadata.tsv
-        self.assertEqual(len(project_zip.namelist()), 19)
+        self.assertEqual(len(project_zip.namelist()), 16)
```

Comment:
We should look at why we are no longer seeing the directories in namelist, that seems odd to me.

We did see a difference of now nesting the library folder in the sample folder, so we should update the commented out file tree to reflect that change.

Reply:
#### A few notes on this
1. There was a bug in the old implementation, where spatial libraries were not nested under their sample folders in the project zip. The bug can be seen below:
  - In `ComputedFile::get_project_single_cell_file` (correct)
```python
    with ZipFile(...) as zip_file:
        ...
        for sample_id, file_paths in sample_to_file_mapping.items():
            for file_path in file_paths:
                # Correct nesting of library data files under sample directory.
                zip_file.write(file_path, Path(sample_id, file_path.name))
```
  - In `ComputedFile::get_project_spatial_file` (bug)
```python
    with ZipFile(...) as zip_file:
        ...
        for sample_id, file_paths in sample_to_file_mapping.items():
            sample_path = project.get_sample_input_data_dir(sample_id)
            for file_path in file_paths:
                # Using relative_to here removes the intended nesting by excluding sample directory.
                zip_file.write(file_path, Path(file_path).relative_to(sample_path))
```
2. The only place that subdirectories seem to be included as their own entries in `ZipFile.namelist()` is during testing of the spatial computed file. In the new implementation with the correction of the above bug, the subdirectories are now no longer included as their own entries in `ZipFile.namelist()`.

Reply:
3. While the tests include in them a commented out file tree (with the number of files supposed to be included and a mention of `(including subdirectory names)`, all of the numbers which are actually asserted differ from the number in the comment. For example:
- `test_single_cell_metadata`
  - `There are 14 files (including subdirectory names)`
   *This comment is incorrect because there are 16 files including subdirectories*
  - `self.assertEqual(len(project_zip.namelist()), 14)`
- `test_multiplexed_metadata`
  - `There are 14 files (including subdirectory names)`
  - `self.assertEqual(len(project_zip.namelist()), 12)`
   *Notice how we test for 12 files, which excludes 2 subdirectories*
- `test_merged_project_anndata_cite_seq`
  - `There are 6 files (including subdirectory names)`
  - `self.assertEqual(len(project_zip.namelist()), 8)`
   *Notice how we test for 8 files, which excludes 3 subdirectories*
- `test_merged_project_anndata_no_cite_seq`
  - `There are 8 files (including subdirectory names)`
  - `self.assertEqual(len(project_zip.namelist()), 10)`
   *Notice how we test for 10 files, which excludes 3 subdirectories*

Reply:
Ok, as long as the comments have all been updated we should be good.